### ISSUE TITLE
How can i find the overlapping data of INSPECT from RSNA and Radfusion ?

### ISSUE THREAD

**Issue by Akshay1-6180** [som-shahlab/INSPECT_public#5]
Based on the cohort data it seems that there is some inhouse data that has been used for the split , and after getting the data its very hard to identify which of the inspect data is RSNA or Radfusion due to which creating the train val split is very confusing, 
Any help regarding this would be helpful 
At the same time providing the split for the INSPECT data would also be valuable

**Comment by ZepengHuo** [comment]
Thanks for the questions.
- What's the need for RSNA/Radfusion provenance? We have made sure training set from those datasets are not in our val/test set, and vice versa (the val/test from them are not in INSPECT training set)

> No training data from RSPECT and Radfusion are included in our validation/test sets. [https://arxiv.org/pdf/2311.10798#page=4.5](url)

- We are working on adding the train/val/test split information + a number of other metadata, e.g. image_datatime, impression_datetime into the dataset. Please stay tuned

**Comment by Akshay1-6180** [comment]
Thank you for your response and for putting together the Stanford dataset—it's greatly appreciated. 
My goal is to train models on RSNA and RadFusion data combined with the standford data while ensuring a proper train/val/test split, keeping RSNA and RadFusion test sets separate. Since the original RSNA dataset lacks a validation set, I created my own, and I want to avoid data contamination by identifying overlapping data. So currently i used a similarity match algorithm to identify the overlapping ones. 
the rsna data was downloaded from here : https://www.kaggle.com/competitions/rsna-str-pulmonary-embolism-detection
radfusion data from here : https://stanfordaimi.azurewebsites.net/datasets/3a7548a4-8f65-4ab7-85fa-3d68c9efc1bd

Any guidance on this would be highly valuable. Using wasserstein distances and taking the 'num_slices', 'mean', 'median', 'min', 'max', 'std_dev', 'sum', 'zero_count', 'max_count', 'min_count' for each ct scan and doing similarity matching with each other , i was able to identify 
1. radfusion - 402/772 
2. rsna - 575/579 

rest i was not able to so thats why any help would be appreciated 

[stanford_rsna_overalapping.csv](https://github.com/user-attachments/files/18688563/stanford_rsna_overalapping.csv)
[stanford_radfusion_overalapping.csv](https://github.com/user-attachments/files/18688564/stanford_radfusion_overalapping.csv)

**Comment by ZepengHuo** [comment]
Thanks for the suggestions. The request is in queue. Stay tuned for update

**Comment by ZepengHuo** [comment]
All the relevant metadata files are published at [AIMI website](https://stanfordaimi.azurewebsites.net/datasets/151848b9-8b31-4129-bc25-cefdf18f95d8)

**Comment by Akshay1-6180** [comment]
@ZepengHuo we have noticed around 500 cases where the nlp is wrong when looking at the impressions

**Comment by ZepengHuo** [comment]
@Akshay1-6180 can you specify `'nlp is wrong'`? Example?

**Comment by Akshay1-6180** [comment]
some examples , gt nlp is ur output and pe_present was using llm , we are currently working with rads to get the correct gt using llms and veriying it , if its fine can attach the mismatched data csv here.
1. image_id:  PE4525fe7
    pe_present:  0
    pe_gt_nlp:  1
    impressions:  IMPRESSION: 1.  No filling defects in the right pulmonary arterial system to suggest pulmonary embolism. Suboptimal opacification of the atretic left pulmonary arterial system due to contrast mixing. 2.  Patchy groundglass opacities throughout the right lung, with new centrilobular opacities in the right middle lobe, can represent an infectious process. 3.  Status post left upper lobectomy. Grossly unchanged appearance of severe bronchiectasis and complete atelectasis of the left lower lobe. The 2.7 cm cavitary lesion in the left upper chest with layering debris is unchanged. There are no substantial differences between the preliminary results and the impressions in this final report.  
    pe_json:  []

2. image_id:  PE452826d
    pe_present:  0
    pe_gt_nlp:  1
    impressions:  IMPRESSION:
    1.  No new filling defects. Resolution of prior pulmonary emboli in the inferior lingular and subsegmental right basilar branches.
    2.  Stable 6mm perifissural right lower lobe nodule since <DATE>.
    pe_json:  []

3.image_id:  PE9f481c
    pe_present:  1
    pe_gt_nlp:  0
    impressions:  IMPRESSION: 1.  STATUS POST AORTIC VALVE REPLACEMENT, ASCENDING AORTIC REPAIR, AND MITRAL VALVE ANNULOPLASTY, AS DESCRIBED ABOVE.  STANDARD POSTSURGICAL CHANGES ARE IDENTIFIED WITHIN THE MEDIASTINUM, AS DESCRIBED. 2.  STATUS POST CABG, WITH LIMA-LAD AND REVERSE SAPHENOUS VEIN GRAFT TO DISTAL RIGHT CORONARY ARTERY, AS DESCRIBED ABOVE. 3.  PULMONARY EMBOLI WITHIN THE PULMONARY ARTERY SUPPLYING THE ANTERIOR APICAL PORTION OF THE RIGHT LUNG, AS WELL AS THE ANTERIOR BASAL PORTION OF THE RIGHT UPPER LOBE, AS DESCRIBED ABOVE. 4.  MILD INFLAMMATORY STRANDING SURROUNDING THE VISUALIZED PORTIONS OF THE PANCREAS.  IN THE APPROPRIATE CLINICAL SETTING, THIS MAY BE SEEN WITH PANCREATITIS.  CLINICAL CORRELATION IS ADVISED. 5.  SMALL BILATERAL PLEURAL EFFUSIONS. 6.  REDEMONSTRATION OF INFEROAPICAL LEFT VENTRICULAR ANEURYSM, AS DESCRIBED. 7.  GYNECOMASTIA. 8.  HYPODENSE LESION WITHIN THE DOME OF THE LIVER.  THIS IS NOT COMPLETELY EVALUATED ON THE CURRENT EXAMINATION.  AS CLINICALLY INDICATED, A SONOGRAM OF THE LIVER MAY BE USEFUL TO FURTHER EVALUATE THIS FINDING. THESE FINDINGS WERE DISCUSSED WITH THE PA-C CARING FOR THIS PATIENT AT <TIME> ON <DATE>. SUMMARY:4-POSSIBLE SIGNIFICANT FINDINGS, MAY NEED ACTION 
    pe_json:  [{'arterial_level': 'segmental', 'lobe_present': {'RUL': ['anterior', 'apical']}}, {'arterial_level': 'segmental', 'lobe_present': {'RUL': ['anterior basal']}}]

**Comment by ZepengHuo** [comment]
Thanks. Can you provide the 500 `image_id` with the incorrect labels? There might be a good chance the NLP labeler is not spitting out correctly

**Comment by Akshay1-6180** [comment]
[mismatch_impression_ids.csv](https://github.com/user-attachments/files/21554172/mismatch_impression_ids.csv)
I have attached the csv @ZepengHuo , sorry for the delay , was checking with the rads on this too , moreover one of the downloaded files PE452890e.nii.gz seems to be corrupted , could u check it form ur end , thanks

**Comment by Akshay1-6180** [comment]
@ZepengHuo is it possible to also provide the INSPECT ids matching to the RSNA and Radfusion since , i have identified that in the mismatch 
provenance
INSPECT      495
rsna          37
radfusion     19

so based on this 56 from rsna and radfusion have wrong labels , since it dont have study id mappings its kind of hard to see which gt would be accurate , since i am mixing rsna , radfusion and inspect and from inspect removing the rsna and radfusion ids , if i know the mapping , i could correct the gt then
